src/components/PaymentMethodChooser.js

```diff
@@ -67,29 +67,31 @@ class PaymentMethodChooser extends React.Component {
     });
   }
 
-  UNSAFE_componentWillReceiveProps(newProps) {
+  componentDidUpdate(prevProps) {
     // TODO: Remove these hacky fixes
     // Most likely means need to rework the logic split between this and SubscriptionCard component
 
-    const { paymentMethodInUse, paymentMethodsList, editMode } = newProps;
+    const { paymentMethodInUse, paymentMethodsList, editMode } = this.props;
 
-    if (!paymentMethodInUse.name) {
-      // set state to modified
-      this.setState({ modified: true });
-      // If there was an existing card, select that
-      if (paymentMethodsList.length > 0) {
-        this.handleChange({ uuid: paymentMethodsList[0].uuid });
+    if (!prevProps.paymentMethodsList && paymentMethodsList) {
```

Comment:
Why introducing this test?

Reply:
To avoid infinite loops, since we're setting state in `componentDidUpdate`. This ensures that the state will only be updated for this condition.